app/uk/gov/hmrc/gform/instructions/InstructionsRenderingService.scala

```diff
@@ -225,13 +226,14 @@ class InstructionsRenderingService(
       new govukSummaryList()(SummaryList(slr :: Nil, "govuk-!-margin-bottom-5"))
     }
 
-    def instructionOrder(bracket: Bracket[Visibility]): Int =
-      bracket
-        .fold(_.source.page.instruction.flatMap(_.order))(_.source.page.instruction.flatMap(_.order))(
-          _.source.instruction.flatMap(_.order))
-        .getOrElse(Integer.MAX_VALUE)
+    def instructionOrder(i: Option[Instruction]): Int = i.flatMap(_.order).getOrElse(Integer.MAX_VALUE)
+    def pageInstructionOrder(p: Page[_]): Int = instructionOrder(p.instruction)
```

Comment:
Why are you using `Page[_]` rather than explicit `Page[Visibility]`?

Reply:
Added [Visibility]

Reply:
Ok, this is weird by itself that it sorts by source, but that is different question altogether.